src/utils/redirectMap.js

```diff
@@ -0,0 +1,64 @@
+import React, { createContext, useContext, useEffect, useState } from "react";
+import io from "socket.io-client";
+
+const SESSION_STORAGE_REDIRECT_MAP_KEY = "nearSocialVMredirectMap";
+
+export function useRedirectMap() {
+  const enableHotReload = useContext(HotReloadContext);
+
+  const [hotReload, setHotReload] = useState(enableHotReload);
+  const [devJson, setDevJson] = useState({
+    components: {},
+    data: {},
+  });
+
+  useEffect(() => {
+    (async () => {
+      if (hotReload) {
+        const socket = io(process.env.BOS_LOADER_WS || "ws://127.0.0.1:8080", {
```

Comment:
Is `process.env` available here in the browser context?

Also, must it be only 127.0.0.1? If developing in github codespaces, you get dedicated host names with https.

Reply:
Fixed here: https://github.com/NEARBuilders/near-bos-webcomponent/commit/d1a2839f63473b59cd49b85fe7ea9384eb439f3b

Reply:
Fixed in this commit: https://github.com/NEARBuilders/near-bos-webcomponent/pull/26/commits/e1af6c47d1a6c67e9afaea9c14c0d4ec3665e26b